nilearn/_utils/compat.py

```diff
@@ -69,10 +69,7 @@ def md5_hash(string):
         return m.hexdigest()
 
 
-if sklearn.__version__ < '0.21':
```

Comment:
This is the only potential sticking point, in my mind -- is this effectively bumping up the sklearn version ?

Reply:
No. What I have done is, irrespective of the version of Scikit-Learn installed, Nilearn always uses the installed Joblib. Scikit-Learn < 0.21 will internally use its own vendored Joblib, independent of Nilearn code.

Unless there is some change in how the vendored Joblib handles its functionality vs the Joblib installed in the library, which I did not notice in the runtimes, this seems to be working.